source/grid/grid_generator.cc

```diff
@@ -777,20 +777,20 @@ namespace GridGenerator
       {
       case 1:
         for (unsigned int x=0; x<=n_subdivisions[0]; ++x)
-          points.push_back (Point<dim> (x*delta[0]));
+          points.push_back (Point<dim> (double(x)*delta[0]));
```

Comment:
Just to help me understand the chain of EnableIfScalar and ProductType: You need to cast x to double here because ProductType of unsigned int and double does not exist, right? Because unsigned int passes the EnableIfScalar declaration.